### DISCUSSION TITLE
Column and Row Span and Other Features in Data Table

### DISCUSSION THREAD

**Discussion by Shriinivas** [Textualize/textual#4754] [Q&A]
I was thinking of enhancing this fantastic Widget to include column and row span and some other stuff, so that it would be easier to build a functional spreadsheet app using it. Just wanted to check if there is any work going on around this already, or should I simply take the existing widget code (_data_table.py) as the basis. 
Also, any specific suggestions for implementation would be really appreciated. Was thinking of creating an embedded `Cell` class with `row_span` and `column_span` attributes, then make changes in `render_cell`, `get_cell_region`, `add_row` etc. Have already tried out kind of "proof of concept" code for column span with some very basic test cases. This involves using isinstance at multiple places, please share if you have any better ideas.

**Comment by darrenburns** [comment]
Sorry for the delayed response! There's no work going on for this that I'm aware of :)

**Comment by qlibp** [comment]
> I was thinking of enhancing this fantastic Widget to include column and row span and some other stuff, so that it would be easier to build a functional spreadsheet app using it. Just wanted to check if there is any work going on around this already, or should I simply take the existing widget code (_data_table.py) as the basis. Also, any specific suggestions for implementation would be really appreciated. Was thinking of creating an embedded `Cell` class with `row_span` and `column_span` attributes, then make changes in `render_cell`, `get_cell_region`, `add_row` etc. Have already tried out kind of "proof of concept" code for column span with some very basic test cases. This involves using isinstance at multiple places, please share if you have any better ideas.

Also interested in this enhancement. Any idea on how to build such third party spreadsheet widget?
